src/term/z-form.h

```diff
@@ -31,21 +31,21 @@
 extern uint vstrnfmt(char *buf, uint max, concptr fmt, va_list vp);
 
 /* Simple interface to "vstrnfmt()" */
-extern uint strnfmt(char *buf, uint max, concptr fmt, ...);
+extern uint strnfmt(char *buf, uint max, concptr fmt, ...) __attribute__((format(printf, 3, 4)));
```

Comment:
I don't know if there's compilers that won't understand the __attribute__ part and flag it as an error.  Angband's current way of omitting the __attribute__ parts if the compiler isn't gcc or clang is here, https://github.com/angband/angband/pull/5527 .

Reply:
A similar solution is introduced in h-config.h.
In my solution, the compiler directly replaces `__attribute__(())` with blanks in the macro if `__GNUC__` is not defined.
I have not checked the `__clang__` definition, but since `__attribute__` is strictly a GCC extension and clang also defines `__GNUC__` to support it, I believe that checking only `__GNUC__` is not a problem.

Reply:
Incidentally, MSVC cannot interpret the `__attribute__` and an error will result.